Add naming scheme reference for billing (#26)
vshn/application-catalog-docs: docs/modules/ROOT/pages/reference/billing.adoc

> = Billing

Reviewer: ```suggestion
= Naming Scheme for Billing
```

> 

Reviewer: Also rename the file to `billing-naming.adoc` or something

Author: Done, but I feel like the example query doesn't fit anymore, if we make the page only about the naming.

Reviewer: I know, but I'm fine with leaving it for now until we have a better place.

This page specifies various aspects of the AppCatalog billing.

== Product Naming in the Database

In order to support all future AppCatalog services and its subproducts, we need to define a naming scheme to differentiate between them.
Each AppCatalog service will have multiple variations, let's take PostgreSQL as an example:

* PostgreSQL standalone without SLA
* PostgreSQL standalone with SLA
* PostgreSQL replicated without SLA
* PostgreSQL replicated with SLA
* PostgreSQL CloudSQL
* PostgreSQL RDS

Every permutation will need an identifier, as it will be associated with a product in the https://kb.vshn.ch/appuio-cloud/references/architecture/metering-data-flow.html#_data_model[product dimension] of the billing database.
Also, depending on the cluster where a given service is running, there could be different prices and discounts.

To accommodate for all these variations, the default identifier of each product in the database should be as follows:

 appcat_$servicename_$variation_$sla:$clustername

* `servicename`: The name of the AppCatalog service, in this example PostgreSQL

> * `variation`: The variation of the service, for example standalone or rds.

Reviewer: I would stay consistent with the words we chose in https://git.vshn.net/vshn/docs/products/-/merge_requests/52
```suggestion
* `architecture`: The architecture type of the service, for example `standalone` or `rds`.
```

* `sla`: The name of the applied SLA.
* `clustername`: The name of the cluster where this given instance is provisioned.

> We can add more specific product identifiers for special cases as needed. See the "examples" section of the https://kb.vshn.ch/appuio-cloud/references/architecture/metering-data-flow.html#_system_idea[system idea] for more information about how the matching works.

Reviewer: ```suggestion
More specific product identifiers for special cases are added as needed.
See the "examples" section of the https://kb.vshn.ch/appuio-cloud/references/architecture/metering-data-flow.html#_system_idea[system idea] for more information about how the matching works.
```

== Query Names for Reporting

The query names in the reporting tool should match the naming for the product.
But it should not contain the `clustername` part.
The cluster name is part of the `product` label that should be returned by the query.

Examples for reporting query names:

* appcat_postgresql_standalone_besteffort

> * appcat_redis_cluster_sla

Reviewer: Maybe use the SLA name? (future-proof if there's more than one SLA option)
```suggestion
* appcat_redis_cluster_guaranteed
```

Author: Right, that was actually my intention :D See the `besteffort` example above it.
